### DISCUSSION TITLE
Error: Response is constant across permutations. Please check your model -> pada LIME explanation

### DISCUSSION THREAD

**Discussion by rajapalawija** [teamalgoritma/community#645] [Q&A] [answered]
Mohon bantuannya.

Saya sebelumnya tidak menemukan error seperti di atas. Namun, pada saat memindahkan project ke device lain, mendapatkan error tersebut. 

Saya membuat function untuk merubah "naiveBayes" menjadi "classification"
<img width="1104" alt="image" src="https://user-images.githubusercontent.com/123149453/234500127-a390d706-f404-487c-85a6-a4c2f035e879.png">

lalu down sampling + mengambil text data
<img width="1104" alt="image" src="https://user-images.githubusercontent.com/123149453/234500159-cff8a19f-6db3-4a50-b023-48a7cb84dcf7.png">

membuat explainer
<img width="1104" alt="image" src="https://user-images.githubusercontent.com/123149453/234500238-5f7e36bf-5683-4585-901f-80601c94b970.png">
dan hasilnya sudah sesuai yaitu class = "text_explainer" "explainer" "list" 

Namun pada saat run di bawah, akan mendapatkan error :
<img width="806" alt="image" src="https://user-images.githubusercontent.com/123149453/234600491-367f30e8-9888-4383-86e3-6a06309c4fb3.png">

Error: Response is constant across permutations. Please check your model

**Comment by amaliasf** [comment, marked as answer]
Dear Pak @rajapalawija,

Terimakasih untuk pertanyaannya Pak.

Terkait error yang bapak hadapi, salah satu alasan hal tersebut umum terjadi dikarenakan input untuk model kita adalah document-term matrix, dengan banyak nilai sparse/kosong. LIME terkadang tidak bekerja dengan baik apabila input mengandung banyak sparse value. Sehingga hal ini sangat sering terjadi dalam kasus text classification. Untuk solusi terkait case ini masih belum terdefinisikan dengan akurat, seperti yang disampaikan pada link https://algoritmaonline.com/text-lime/

Namun kami juga menemukan referensi yang menyatakan bahwa error tersebut berhasil di atasi, boleh silahkan dicoba solusi https://www.appsloveworld.com/r/100/604/lime-library-in-r-throwing-error-response-is-constant-across-permutations-plea

Demikian yang dapat kami sampaikan, semoga membantu Pak

**Reply by rajapalawija** [reply]
Terima kasih mbak @amaliasf 

Ini berhasil tp lead ke masalah lain. Saya coba resolve sendiri dulu
